core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/AddChildProfileViewModel.kt

```diff
@@ -0,0 +1,235 @@
+package com.ustadmobile.core.viewmodel.person.child
+
+import com.ustadmobile.core.MR
+import com.ustadmobile.core.account.LearningSpace
+import com.ustadmobile.core.db.UmAppDatabase
+import com.ustadmobile.core.impl.UstadMobileSystemCommon
+import com.ustadmobile.core.impl.appstate.ActionBarButtonUiState
+import com.ustadmobile.core.impl.nav.UstadSavedStateHandle
+import com.ustadmobile.core.util.ext.appendSelectedAccount
+import com.ustadmobile.core.util.ext.onActiveEndpoint
+import com.ustadmobile.core.view.UstadView
+import com.ustadmobile.core.viewmodel.UstadEditViewModel
+import com.ustadmobile.core.viewmodel.clazz.list.ClazzListViewModel
+import com.ustadmobile.door.util.systemTimeInMillis
+import com.ustadmobile.lib.db.entities.Person
+import com.ustadmobile.lib.db.entities.PersonParentJoin
+import io.github.aakira.napier.Napier
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.builtins.ListSerializer
+import org.kodein.di.DI
+import org.kodein.di.instance
+
+
+data class AddChildProfileUiState(
+    val onAddChildProfile: String? = null,
+    val childProfiles: List<Person> = emptyList(),
+    val personParenJoinList: List<PersonParentJoin> = emptyList(),
+    val profileIncludingParentAndChildren: List<Person> = emptyList(),
```

Comment:
This looks lke we would be storing the Person object for the child twice; we don't want to do that because it leads to potential inconsistency. Should have a single point of truth. Why not have a val for parentPerson?